### DISCUSSION TITLE
Have you heard of OpenHarmony or HarmonyOS ?

### DISCUSSION THREAD

**Discussion by chatterzhao** [beeware/toga#2881] [Polls]
- [OpenHarmony](https://gitee.com/openharmony)
- [HarmonyOS](https://www.harmonyos.com)

**Comment by chatterzhao** [comment]
The relationship between open-source OpenHarmony and HarmonyOS is similar to the relationship between AOSP and Android.

Huawei has now launched a new version, currently called HarmonyOS NEXT, which has removed AOSP and no longer supports Android applications.

In China, Chinese documentation is updated first. It is recommended to use browser translation instead of using the language switch on the documentation.
-  [OpenHarmony](https://www.openharmony.cn/mainPlay)
- [HarmonyOS NEXT](https://developer.huawei.com/consumer/cn/doc/harmonyos-releases-V5/next-dev-beta-rn-V5?catalogVersion=V5)

> [HarmonyOS API](https://developer.huawei.com/consumer/cn/doc/harmonyos-references-V5/development-intro-api-V5?catalogVersion=V5)(Kit is API)

The following content is to provide some background information. There's nothing like a to-do list here.

HarmonyOS is the system used by Huawei, a Chinese company. After Huawei was unable to use the Android system in 2019, they introduced their backup system called HarmonyOS. The version Huawei contributed to open-source organizations is called OpenHarmony. Their relationship is similar to that of Android and AOSP.

Previously, HarmonyOS included AOSP for transition purposes, allowing Android applications to be installed seamlessly. However, Huawei has now launched a new version, currently called HarmonyOS NEXT, which has removed AOSP and no longer supports Android applications.

They have gone their separate ways, and due to Huawei's large market share in China, they seem to be able to do this successfully. After taking this approach, top applications in China have started redeveloping. It's said that the first batch includes over 4,000 top applications, satisfying 99% of usage needs.

Some users are already in the beta testing phase, and progress is going smoothly. Public beta testing will begin on October 8, 2024, when it will be pushed to mobile users. It's expected to be officially released in November along with the Mate 70. News reports suggest that previously released phones will also support upgrading to HarmonyOS NEXT. At this point, developers in China will need to develop applications for an entirely new platform, similar to iOS, Android, and HarmonyOS, creating such a landscape.
HarmonyOS is a unique existence. Unlike macOS, iOS, and iPadOS which are categorized separately, it's a single system that simultaneously supports phones, tablets, watches, TVs, HiCar, and soon, PCs.

Additional information: In terms of mobile devices alone, HarmonyOS has already surpassed Apple's iOS market share in China, where Apple phones have a significant presence.

It's important to note a distinction: NEXT is a new version, not a new system. OpenHarmonyOS and HarmonyOS have already had five versions: 1, 2, 3, 4, and 5 (also known as NEXT). The versions of OpenHarmonyOS and HarmonyOS are synchronized. They now have 900 million devices using their system.

**Comment by NeverYouGetYouUp001** [comment]
I think HarmonyOS NEXT is some kind of Linux kernel,do we should add toga into HarmonyOS NEXT?they have a way to side load app like iOS.i think let developer to develop a app to HarmonyOS NEXT is easy.especially like open source develop do that thing.

**Comment by lisonge** [comment]
At their developer conference, Huawei executives claimed that the kernel of their HarmonyOS surpasses the Linux kernel, offering better security and smoother performance, with a 10.7% improvement in performance by comparison

![img](https://img-s-msn-com.akamaized.net/tenant/amp/entityid/BB1oDbvd.img?w=640&h=342&m=6)

<https://www.bilibili.com/video/BV1rs421T7k9/?t=83>

**Comment by mhsmith** [comment]
See https://github.com/beeware/toga/issues/2880#issuecomment-2389794064.
